clinvar/src/main/scala/org/broadinstitute/monster/etl/clinvar/ClinvarPipeline.scala

```diff
@@ -34,107 +33,83 @@ object ClinvarPipeline {
-    // Further split the VCV stream to create a releases table.
-    val vcvBranches = VcvBranches.fromVcvStream(vcvs)
-    // Further split the SCV stream to create new submitter and submission entities.
-    val scvBranches = ScvBranches.fromScvStream(scvs)
+    // De-duplicate where needed.
+    val uniqueGenes = archiveBranches.genes.distinctBy(_.id)
+    val uniqueSubmitters = archiveBranches.submitters.distinctBy(_.id)
+    val uniqueSubmissions = archiveBranches.submissions.distinctBy(_.id)
 
     // Write everything back to storage.
     MsgIO.writeJsonLists(
-      variations,
+      archiveBranches.variations,
       "Variations",
       s"${parsedArgs.outputPrefix}/variation"
     )
     MsgIO.writeJsonLists(
-      geneBranches.genes,
+      uniqueGenes,
       "Genes",
       s"${parsedArgs.outputPrefix}/gene"
     )
     MsgIO.writeJsonLists(
-      geneBranches.geneAssociations,
+      archiveBranches.geneAssociations,
       "Gene Associations",
       s"${parsedArgs.outputPrefix}/gene_association"
     )
     MsgIO.writeJsonLists(
-      vcvBranches.vcvs,
+      archiveBranches.vcvs,
       "VCVs",
       s"${parsedArgs.outputPrefix}/variation_archive"
     )
     MsgIO.writeJsonLists(
-      vcvBranches.vcvReleases,
+      archiveBranches.vcvReleases,
       "VCV Releases",
       s"${parsedArgs.outputPrefix}/variation_archive_release"
     )
     MsgIO.writeJsonLists(
-      rcvs,
+      archiveBranches.rcvs,
       "RCV Accessions",
       s"${parsedArgs.outputPrefix}/rcv_accession"
     )
     MsgIO.writeJsonLists(
-      scvBranches.scvs,
+      archiveBranches.scvs,
       "SCVs",
       s"${parsedArgs.outputPrefix}/clinical_assertion"
     )
     MsgIO.writeJsonLists(
-      scvBranches.submitters,
+      uniqueSubmitters,
       "Submitters",
       s"${parsedArgs.outputPrefix}/submitter"
     )
     MsgIO.writeJsonLists(
-      scvBranches.submissions,
+      uniqueSubmissions,
       "Submissions",
       s"${parsedArgs.outputPrefix}/submission"
     )
     MsgIO.writeJsonLists(
-      scvVariations,
+      archiveBranches.scvVariations,
       "SCV Variations",
       s"${parsedArgs.outputPrefix}/clinical_assertion_variation"
     )
     MsgIO.writeJsonLists(
-      scvObservations,
+      archiveBranches.scvObservations,
       "SCV Observations",
       s"${parsedArgs.outputPrefix}/clinical_assertion_observation"
     )
     MsgIO.writeJsonLists(
-      scvTraitSets,
+      archiveBranches.scvTraitSets,
       "SCV Trait Sets",
       s"${parsedArgs.outputPrefix}/clinical_assertion_trait_set"
     )
     MsgIO.writeJsonLists(
-      scvTraits,
+      archiveBranches.scvTraits,
       "SCV Traits",
       s"${parsedArgs.outputPrefix}/clinical_assertion_trait"
     )
-    MsgIO.writeJsonLists(
-      traitSets,
-      "Trait Sets",
-      s"${parsedArgs.outputPrefix}/trait_set"
-    )
-    MsgIO.writeJsonLists(
-      traits,
-      "Traits",
-      s"${parsedArgs.outputPrefix}/trait"
-    )
```

Comment:
I wiped the VCV trait stuff for now. As I was digging into the data I had even more ??? moments, so I think it'll be good to look at it fresh.